.changelog/19490.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+client: fix a bug where task level Consul hook never gets called
```

Comment:
For the changelog, I would word this in terms of the user-facing impact rather than the internal code ("hooks" aren't something we document). We fixed a bug where the environment variable / file for the Consul token weren't written.

Reply:
rephrased in 2fbde1e9d0af0812fdaca059ca4d417c60de1757